.circleci/task_lint.sh

```diff
@@ -8,9 +8,11 @@ set -o pipefail
 
 echo "Check license header..."
 python3 scripts/lint/check_license_header.py HEAD~1
```

Comment:
I have a side question: why is it `HEAD~1` not `HEAD`? Why do we check the current commits first parent

Reply:
`HEAD` is the current commit. Basically we only want to check those files that are modified. For those unmodified files, we know they are already well-formatted.

Reply:
That makes sense, thanks!